src/middlewared/middlewared/api/v25_04_0/catalog.py

```diff
@@ -0,0 +1,102 @@
+from pydantic import ConfigDict, Field, HttpUrl
+
+from middlewared.api.base import BaseModel, ForUpdateMetaclass, NonEmptyString, single_argument_args
+
+
+__all__ = [
+    'CatalogEntry', 'CatalogUpdateArgs', 'CatalogUpdateResult', 'CatalogTrainsArgs', 'CatalogTrainsResult',
+    'CatalogSyncArgs', 'CatalogSyncResult', 'CatalogAppInfo', 'CatalogAppsArgs', 'CatalogAppsResult',
+    'CatalogAppDetailsArgs', 'CatalogAppDetailsResult',
+]
+
+
+class CatalogEntry(BaseModel):
+    id: NonEmptyString
+    label: NonEmptyString = Field(pattern=r'^\w+[\w.-]*$')
+    preferred_trains: list[NonEmptyString]
+
+
+@single_argument_args('catalog_update')
+class CatalogUpdateArgs(BaseModel, metaclass=ForUpdateMetaclass):
+    preferred_trains: list[NonEmptyString]
+
+
+class CatalogUpdateResult(BaseModel):
+    result: CatalogEntry
+
+
+class CatalogTrainsArgs(BaseModel):
+    pass
+
+
+class CatalogTrainsResult(BaseModel):
+    result: list[NonEmptyString]
+
+
+class CatalogSyncArgs(BaseModel):
+    pass
+
+
+class CatalogSyncResult(BaseModel):
+    result: None
+
+
+class Maintainer(BaseModel):
+    name: str
+    email: str
+    url: HttpUrl | None
+
+
+class CatalogAppInfo(BaseModel):
+    app_readme: str = Field(description='HTML content of the app README.')
```

Comment:
`null=True` in the old schema implies that this should be `str | None`, same for several other fields in this model.